src/kOS/Suffixed/Addons.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Encapsulation;
+
+namespace kOS.Suffixed
+{
+    public class AddonList : Structure
+    {
+        private Addon KACAddon;
+        private Addon RTAddon;
+        private Addon AGXAddon;
+
+        public AddonList()
+        {
+            KACAddon = new Addon ("KAC");
+            RTAddon = new Addon ("RT");
+            AGXAddon = new Addon ("AGX");
+
+            InitializeSuffixes();
+        }
+
+        private void InitializeSuffixes()
+        {
+            AddSuffix("KAC", new Suffix<Addon>(() => KACAddon));
+            AddSuffix("RT", new Suffix<Addon>(() => RTAddon));
+            AddSuffix("AGX", new Suffix<Addon>(() => AGXAddon));
+        }
+
+        public override string ToString()
+        {
+            return string.Format("{0} AddonList", base.ToString());
+        }
+    }
+
+    public class Addon : Structure
+    {
+        public string addonName = "";
+
+        public Addon(string name)
+        {
+            addonName = name;
+
+            InitializeSuffixes();
+        }
+        private void InitializeSuffixes()

```

Comment:
Conditional suffixes that change per instance of the object are going to make syntax highlighting data impossible to export, and pretty much means we're deciding to abandon that idea.